Foliage2

kinsch opened the pull request:
Created new folder for Playermodels and added investigator model.

jjszaniszlo committed:
add FirstPersonCamera.cs

jjszaniszlo committed:
add script to player scene

jjszaniszlo committed:
scale terrain

jjszaniszlo committed:
- export constants so they are editable within the editor
- add a movement follow direction to have the direction follow arbitrary nodes (future proofing)
- change input to follow both a follow target or world space.
- change input to utilize gameplay related input mappings within the editor

jjszaniszlo committed:
- add basic camera rotation
- add exported settings for editor configuration
- add escape as a temporary way to esacape mouse capture mode

jjszaniszlo committed:
- add camera script to Head node

jjszaniszlo committed:
- add new gameplay input maps for both wasd and arrow keys

jjszaniszlo committed:
add tree models

jjszaniszlo committed:
add bush textures

jjszaniszlo committed:
add godot-git-plugin textures

jjszaniszlo committed:
add terrain textures

jjszaniszlo committed:
add tree textures

jjszaniszlo committed:
add basic terrain material

jjszaniszlo committed:
add trees and canvas shader to world scene

jjszaniszlo committed:
create stylized shader

jjszaniszlo committed:
move textures to assets

jjszaniszlo committed:
add bush and tree scenese

jjszaniszlo committed:
change resolution

jjszaniszlo committed:
add main scene script for environment

jjszaniszlo committed:
adds shader material to world scene graph

jjszaniszlo committed:
add sky shader

jjszaniszlo committed:
change formatting on firstpersoncamera.cs

jjszaniszlo committed:
add stars image

jjszaniszlo committed:
add collision to tree

m35691 committed:
imported tree 36

m35691 committed:
(demo) adding foliage

jjszaniszlo committed:
move stars to assets folder

jjszaniszlo committed:
Update README.md

(cherry picked from commit 26fcf53143252f501bf32a95270131d17ecd4af6)

jjszaniszlo committed:
change solution name to NightmareNegotiations.sln

jjszaniszlo committed:
change viewport modes

jjszaniszlo committed:
create User.cs

jjszaniszlo committed:
Create popup scene

jjszaniszlo committed:
Create PopUp.cs, which is the script file for the PopUp scene

jjszaniszlo committed:
Create Packet.cs, the datastructures representing packets.

jjszaniszlo committed:
create MainMenu scene.

jjszaniszlo committed:
Create MainMenu.cs, the script for main menu scene.

jjszaniszlo committed:
create main scene, the start point for the game

jjszaniszlo committed:
create Main.cs, the script for the main scene

jjszaniszlo committed:
Create Client.cs, the communication method for clients to interact with the relay server.

jjszaniszlo committed:
change solution name to NightmareNegotiations.sln

jjszaniszlo committed:
change transform of directional light 3D

jjszaniszlo committed:
.idea solution name change

jjszaniszlo committed:
add EventHandler postfix to all signal related delegates

jjszaniszlo committed:
fix typo for WebRtcMultiplayerPeer variable

jjszaniszlo committed:
change all signal delegates to be accessed publicly since it's required for them to be used within the editor.

jjszaniszlo committed:
add world scene template

jjszaniszlo committed:
add signal receiver for the SinglePlayer button on the Main Menu

jjszaniszlo committed:
change project main scene to main.tscn

jjszaniszlo committed:
add main menu as a child to the main root menu

jjszaniszlo committed:
remove username entry from main menu

jjszaniszlo committed:
add lobby scene

jjszaniszlo committed:
add UsernamePrompt scene and script

jjszaniszlo committed:
instantiate username prompt instead of main menu

jjszaniszlo committed:
add refresh.png, the refresh icon

jjszaniszlo committed:
add lobby selection menu scene

jjszaniszlo committed:
add script for lobby selection menu scene

jjszaniszlo committed:
- add signal receivers for all buttons
- add implementation for back button

jjszaniszlo committed:
move world and Main to better locations and update names

jjszaniszlo committed:
update namespace for Main

jjszaniszlo committed:
add basic rtcPeer connection boilerplate

jjszaniszlo committed:
add User to globals

jjszaniszlo committed:
add comment explanation

jjszaniszlo committed:
make data structure references readonly

m35691 committed:
More Foliage

jjszaniszlo committed:
move back

jjszaniszlo committed:
move back

jjszaniszlo committed:
Merge remote-tracking branch 'origin/demo' into demo

jjszaniszlo committed:
Merge pull request #9 from kinsch/demo

Kenzie Fork ;D

jjszaniszlo committed:
change sampling from linear to nearest on trees 29-36

jjszaniszlo committed:
editor forced change removing uid from bush02 for some reason

jjszaniszlo committed:
move world and its script to its own scene directory

jjszaniszlo committed:
Match folder names better to match newer conventions i made within the old scenes folder

jjszaniszlo committed:
move back SourceCode so that C# respects namespace

jjszaniszlo committed:
remove unused import on Client.cs

jjszaniszlo committed:
add contribution guidelines to README.md

jjszaniszlo committed:
move scene scripts to src directory

jjszaniszlo committed:
added some of the basic packet parsing types.  TODO: polymorphic json parsing using C# reflection.

jjszaniszlo committed:
Merge branch 'main' into demo

jjszaniszlo committed:
fix assets being upper case

m35691 committed:
Changed assets, materials, scenes, and shaders folders to uppercase.

jjszaniszlo committed:
Merge pull request #11 from kinsch/Foliage2

Changed assets, materials, scenes, and shaders folders to uppercase.

jjszaniszlo committed:
add tree to world

jjszaniszlo committed:
Merge branch 'demo' of https://github.com/jjszaniszlo/kasm into demo

jjszaniszlo committed:
Merge branch 'demo' of https://github.com/jjszaniszlo/kasm into demo

jjszaniszlo committed:
Merge remote-tracking branch 'origin/demo' into demo

jjszaniszlo committed:
revert file name changes

jjszaniszlo committed:
revert file name changes

jjszaniszlo committed:
Merge branch 'demo' of https://github.com/jjszaniszlo/kasm into demo

jjszaniszlo committed:
add dlls

jjszaniszlo committed:
revert file names (hopefully fixed this time)

jjszaniszlo committed:
add these dlls to ignore

jjszaniszlo committed:
remove lowercase assets

jjszaniszlo committed:
Delete assets directory

jjszaniszlo committed:
Delete materials directory

jjszaniszlo committed:
Delete scenes directory

jjszaniszlo committed:
Delete shaders directory

jjszaniszlo committed:
remove godot-git-plugin (causes more issues than solves)

jjszaniszlo committed:
remove godot-git-plugin (causes more issues than solves)

m35691 committed:
test

m35691 committed:
test

jjszaniszlo committed:
Merge branch 'demo' of https://github.com/jjszaniszlo/nightmare-negotiations into demo

m35691 committed:
Merge branch 'demo' of https://github.com/jjszaniszlo/nightmare-negotiations into Foliage2

jjszaniszlo committed:
Begin adding a pause menu

jjszaniszlo committed:
- start RtcClient code
- add Globals.cs boilerplate.

jjszaniszlo committed:
remove reference to the git plugin which was removed

m35691 committed:
Merge branch 'demo' of https://github.com/kinsch/kasm into demo

m35691 committed:
Merge branch 'Foliage2' of https://github.com/kinsch/kasm into demo

jjszaniszlo committed:
add IClient.cs

jjszaniszlo committed:
Add IServer.cs

jjszaniszlo committed:
Initialize either Client or Server depending on os feature "dedicated_server"

jjszaniszlo committed:
change namespace to reflect location

jjszaniszlo committed:
Add Message.cs

jjszaniszlo committed:
Add MessageType.cs

jjszaniszlo committed:
Clean up RtcClient to prepare for refactor

jjszaniszlo committed:
Create MultiplayerState.cs

jjszaniszlo committed:
Create NetworkManager.cs

jjszaniszlo committed:
Create Peer.cs

jjszaniszlo committed:
Add Global autoloads for both NetworkManager and MultiplayerState

jjszaniszlo committed:
small fixes to Message naming and type naming for the old Packets.

jjszaniszlo committed:
Create WebRtcServer.cs

jjszaniszlo committed:
Create export config

jjszaniszlo committed:
Create multiplayer rtc server

jjszaniszlo committed:
Create scenes for both the client and server.

m35691 committed:
Created folder for Playermodels and added investigator model.

m35691 committed:
Created folder 'PlayerModels' under texture and added investigator texture.

jjszaniszlo commented:
Cool model!

It could honestly also play as a model for one of the NPCs (monsters, etc), because of the aesthetic that it has. 

I would say a good change would be to move it to a folder called "characters" or "Characters", since the previous folder already says the word model :)

So the folder structures would look something like:
Assets/models/characters/investigator/investigator.fbx
Assets/textures/characters/investigator/investigator.png

All in the name of keeping organized

jjszaniszlo committed:
add a tentative file structure specification.

jjszaniszlo committed:
Merge branch 'main' into demo

m35691 committed:
Importing the different fbx files for possible character models.

All fbx files are taken from the character pack 'Adventures Character
Pack' https://kaylousberg.itch.io/.

m35691 committed:
Importing all gltf files from character pack.

Gltf files are taken from the 'Adventurers Character Pack' on
https://kaylousberg.itch.io/. The barbarian, rogue, and rogue_hooded
files have been reimported to have a root type of Character Body 3d and
a root name of 'RootScene', as well as for their animations- all walking
animations and idle animations have been set to a loop mode of 'linear'.

m35691 committed:
Importing all texture files from character pack.

m35691 committed:
text scene files for character objects- barbarian, rogue, and rogue_hooded.

Certain character addons have been hidden for character aesthetic.

m35691 committed:
Investigator model deleted.

jjszaniszlo committed:
Merge branch 'main' into Foliage2

jjszaniszlo committed:
create scenes for global multiplayer related state.

m35691 committed:
Added collision to player models.

m35691 committed:
Merge branch 'Foliage2' of https://github.com/kinsch/kasm into Foliage2

m35691 committed:
scenes/global.tcsn

jjszaniszlo committed:
Merge branch 'demo' into Foliage2

m35691 committed:
Importing all gltf assets for cemetary map.

m35691 committed:
Importing textures for cemetary map.

m35691 committed:
Created new scene for cemetary map (Map1) and began dragging in assets.

m35691 committed:
proj

m35691 committed:
edited map

jjszaniszlo commented:
Too many things diverged since I did all that for demo.  Im going to make a new branch, cherry pick your commits and make a new PR for u.

jjszaniszlo commented:
I finished making a PR with your changes + organizational changes.  I really like what you've done so far!

jjszaniszlo commented:
closing because these changes are merged into Import-Assets already
